assets/app/views/_templateopt.html

```diff
@@ -22,12 +22,16 @@
           type="text"
           placeholder="{{ parameter | parameterPlaceholder }}"
           ng-model="parameter.value"
-          ng-required="parameter.required && !parameter.generate">
+          ng-required="parameter.required && !parameter.generate"
+          ng-trim="false">
     </span>
     <div class="help-block" ng-if="parameter.description">{{parameter.description}}</div>
     <div ng-show="paramForm[paramID].$error.required && paramForm[paramID].$touched" class="has-error">
       <div class="help-block">{{parameter.name}} is required.</div>
     </div>
+    <div ng-show="isOnlyWhitespace(parameter.value)" class="has-warning">
+      <span class="help-block">This parameter value only has spaces.</span>
```

Comment:
Hmm, I feel like we need to be more clear that its ok that they've only entered whitespace and that we are just letting them know...  especially for accessibility since the color isnt enough to distinguish between error and warning in that case.